devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy

```diff
@@ -0,0 +1,674 @@
+				break
+			default:
+				logDebug "Unknown Sensor Type: ${cmd.sensorType}"
+		}
+	}
+	return []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+	logTrace "ConfigurationReport ${cmd}"
+
+	runIn(4, refreshPendingChanges)
+
+	def param = configParams.find { it.num == cmd.parameterNumber }
+	if (param) {
+		def val = cmd.scaledConfigurationValue
+
+		logDebug "${param.name}(#${param.num}) = ${val}"
+		state["configParam${param.num}"] = val
+	} else {
+		logDebug "Parameter #${cmd.parameterNumber} = ${cmd.configurationValue}"
+	}
+	return []
+}
+
+def refreshPendingChanges() {
+	sendEvent(name: "pendingChanges", value: "${pendingChanges} Pending Changes", displayed: false)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	logTrace "NotificationReport: $cmd"
+	def result = []
+
+	if (cmd.notificationType == 0x06 && cmd.event == 0x16) {
+		result << sensorValueEvent(1)
+	} else if (cmd.notificationType == 0x06 && cmd.event == 0x17) {
+		result << sensorValueEvent(0)
+	} else if (cmd.notificationType == 0x07) {
+		if (cmd.event == 0x00) {
+			result << createEvent(descriptionText: "$device.displayName covering was restored", isStateChange: true)
+			cmds = [zwave.batteryV1.batteryGet(), zwave.wakeUpV1.wakeUpNoMoreInformation()]
+			result << response(commands(cmds, 1000))
+		} else if (cmd.event == 0x01 || cmd.event == 0x02) {
+			result << sensorValueEvent(1)
+		} else if (cmd.event == 0x03) {
+			result << createEvent(descriptionText: "$device.displayName covering was removed", isStateChange: true)
+		}
+	} else if (cmd.notificationType) {
+		def text = "Notification $cmd.notificationType: event ${([cmd.event] + cmd.eventParameter).join(", ")}"
+		result << createEvent(name: "notification$cmd.notificationType", value: "$cmd.event", descriptionText: text, displayed: false)
+	} else {
+		def value = cmd.v1AlarmLevel == 255 ? "active" : cmd.v1AlarmLevel ?: "inactive"
+		result << createEvent(name: "alarm $cmd.v1AlarmType", value: value, displayed: false)
+	}
+
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensorbinaryv2.SensorBinaryReport cmd) {
+	logTrace "SensorBinaryReport: $cmd"
+	def map = [:]
+	map.value = cmd.sensorValue ? "open" : "closed"
+	map.name = "contact"
+	if (map.value == "open") {
+		map.descriptionText = "${device.displayName} is open"
+	} else {
+		map.descriptionText = "${device.displayName} is closed"
+	}
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.indicatorv1.IndicatorReport cmd) {
+	logTrace "${cmd}"
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	logDebug "Ignored Command: $cmd"
+	return []
+}
+
+private getEventMap(name, value, displayed=null, desc=null, unit=null) {
+	def isStateChange = (device.currentValue(name) != value)
+	displayed = (displayed == null ? isStateChange : displayed)
+	def eventMap = [
+			name: name,
```

Comment:
Could You indent this section with 2 tabs ?

Reply:
OK, I modified it.